cert/selfsigned.go

```diff
@@ -276,7 +276,7 @@ func GenCertPair(org, certFile, keyFile string, tlsExtraIPs,
 		return err
 	}
 	if err = ioutil.WriteFile(keyFile, keyBuf.Bytes(), 0600); err != nil {
-		os.Remove(certFile)
```

Comment:
Does it mean when the `keyFile` fails at creation we want to remove `certFile`?

Reply:
That makes sense. Actually, looking at this more carefully, if `Open` fails, that is definitely what we want. If `Open` succeeds then we want to delete both. This can only be solved by a transactional write function as suggested on the linked PR or careful error checking. I will close this PR and see about a transactional write function.